### ISSUE TITLE
Add react hooks support

### ISSUE THREAD

**Issue by LosYear** [maslianok/react-resize-detector#108]
Hi! Thank you for your package

I'm wondering, are there any plans to support React hooks? Something like
```js
const {width, height} = useResizeDetector(ref);
```
I believe it can be useful nowadays

**Comment by maslianok** [comment]
Great push, @LosYear :)

I also think it will be a good feature for the library.
I'll try to work on it this month

**Comment by bluenote10** [comment]
There is also:

- [ZeeCoder/use-resize-observer](https://github.com/ZeeCoder/use-resize-observer)
- [envato/react-resize-observer-hook](https://github.com/envato/react-resize-observer-hook)

Perhaps they can serve as an inspiration.

**Comment by maslianok** [comment]
Ok, there are too many likes to this feature request, so I can't procrastinate anymore :)

Today I published v`6.0.0-alpha.0` that supports a hook approach.

Please, give it a shot:

```jsx
import { useResizeDetector } from 'react-resize-detector';

const CustomComponent = () => {
  const { width, height, ref } = useResizeDetector();
  return <div ref={ref}>{`${width}x${height}`}</div>;
};
```

you can pass props as an object to useResizeDetector. For example, `useResizeDetector({ refreshMode: 'debounce', refreshRate: 1000 })`

**Comment by maslianok** [comment]
@bluenote10 thanks for the provided links 👍

**Comment by stefann01** [comment]
Typescript support? Setting ref={ref} in typescripts it says that 

**Type 'MutableRefObject<Element | undefined>' is not assignable to type 'string | ((instance: HTMLDivElement | null) => void) | RefObject<HTMLDivElement> | null | undefined'.
  Type 'MutableRefObject<Element | undefined>' is not assignable to type 'RefObject<HTMLDivElement>'.
    Types of property 'current' are incompatible.
      Type 'Element | undefined' is not assignable to type 'HTMLDivElement | null'.
        Type 'undefined' is not assignable to type 'HTMLDivElement | null'.**

**Comment by maslianok** [comment]
@stefann01 @SteffeyDev thanks guys for reporting/resolving the issues.

Published in v6.0.1-rc.4
